### PR TITLE
add makefile and improve github actions

### PR DESCRIPTION
**Pull request by mmulholla**
Add makefile with standard build tasks

Add go fmt and go mod tidy to build

Extend github actions to be used on Pull requests.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	go fmt -x ./...
+
+.PHONY: bin
+bin:
+	 go build *.go
+	 hack/build.sh
+
+.PHONY: test
+test:
+	 go test ./...
```
Comment (baijum) [inline Makefile:22]: Can we add `-v` flag for verbose output?

Comment (mmulholla) [reply Makefile]: Added.

**Comment by mmulholla** [comment]
Note the build will not pass until this PR is merged: https://github.com/redhat-certification/chart-verifier/pull/51

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	 go mod tidy
+
+.PHONY: gofmt
+gofmt:
+	go fmt -x ./...
+
+.PHONY: bin
+bin:
+	 go build *.go
+	 hack/build.sh
```
Comment (baijum) [inline Makefile:18]: Do we need both these commands? Now there are two binaries, right?

**Comment by mmulholla** [comment]
@baijum all working now - ready for final review

File: Makefile (hunk 2 above)
Comment (mmulholla) [reply Makefile]: True, removed the first.

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,4 @@
 /.idea/
 /out/
+main
```
Comment (baijum) [inline .gitignore:3]: This may not be required as the binary output goes inside the `out` directory.

**Comment by baijum** [comment]
@mmulholla The title still has "DO NOT MERGE". Please change it it's changed.

File: .gitignore (hunk 3 above)
Comment (mmulholla) [reply .gitignore]: removed

**Review by baijum — APPROVED** [review]
LGTM

**Pull request merged**